### PR TITLE
Cardano URI Scheme

### PR DESCRIPTION
**Pull request by SebastienGllmt**
This is a specification of what we've already implemented in both the Yoroi Extension and in Yoroi Mobile

### CODE REVIEW

**Comment by rphair** [comment]
Updated stake pool specific URI scheme CIP in response: https://github.com/cardano-foundation/CIPs/pull/25/commits/0ca54385a0ddf6c398ff283bacb92405f57c212c

**Comment by gufmar** [comment]
> `We should be wary of people who disguise “donate” links as actually opening up a phishing website that LOOKS like a wallet.`

Could it be a security option to let users set a recognition word/phrase inside the wallet?
This is something phishing wallets can't know.

1) one time setup: Please set something you know, such as your preferred color, your birth location, your first car...
2) crucial wallet areas such as landing page for web+Cardano URIs clearly show this locally stored string

I know not perfect but an option

**Comment by SebastienGllmt** [comment]
@gufmar the wallet checksum already plays this role for Yoroi

File: PaymentUrls/PaymentUrls.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+CIP: TODO
```
Comment (KtorZ) [inline PaymentUrls/PaymentUrls.md:2]: :point_up: reminder

File: PaymentUrls/PaymentUrls.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+2. Depending on the protocol registration method (see next section), browsers generally enforce a `web+` or `ext+` prefix for non-whitelisted protocols (note: `bitcoin:` was whitelisted). The prefix `ext+` is recommended for extensions, but not mandatory.
+
+## ABNF Grammar (Proposal)
+
+This is an initial, simplified protocol definition for fast implementation; it only requires an address and an optional amount parameter. As discussed above, these rules are likely to evolve in time in order to support additional features, including unique capabilities of the Cardano blockchain.
+
+```
+cardanourn     = "web+cardano:" cardanoaddress [ "?" amountparam ]
+cardanoaddress = *base58
```
Comment (KtorZ) [inline PaymentUrls/PaymentUrls.md:50]: We probably want bech32 here too right?

File: PaymentUrls/PaymentUrls.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+1. We cannot prompt the user to send the funds right away as they may not be fully aware of the URI they clicked or were redirected to. Instead, it may be better to simply pre-populate fields in a transaction.
+2. We should be wary of people who disguise “donate” links as actually opening up a phishing website that LOOKS like a wallet.
+
+# Rationale
+
+## Why not use Universal links, deep links and other non-protocol-based Solution
+
+An alternative solution to the original problem described above is to use standard URL links in combination with a routing backend system. The routing system is used to redirect to the app's URI. The advantage of this scheme is that it allows to provide a fallback mechanism to handle the case when no application implementing the protocol is installed (for instance, by redirecting to the App Store or Google Play). This is the approach behind iOS Universal Links and Android App Links. In general, it provides a better user experience but requires a centralized system which makes it unsuitable for as a multi-app standard.
+
+## Read More
```
Comment (KtorZ) [inline PaymentUrls/PaymentUrls.md:67]: Maybe add a link to existing Yoroi implementations as a reference?

**Review by KtorZ — APPROVED** [review]

File: PaymentUrls/PaymentUrls.md (hunk 1 above)
Comment (crptmppt) [reply PaymentUrls/PaymentUrls.md:2]: proposing 12

Comment (mmahut) [reply PaymentUrls/PaymentUrls.md:2]: FYI https://github.com/cardano-foundation/CIPs/pull/44#discussion_r538215730

File: PaymentUrls/PaymentUrls.md (hunk 3 above)
Comment (SebastienGllmt) [reply PaymentUrls/PaymentUrls.md:67]: Our implementation is probably too deeply coupled with the rest of our code to serve as a good reference unfortunately

**Review by crptmppt — APPROVED** [review]

**Comment by rphair** [comment]
@SebastienGllmt it is hard to believe this CIP was put through without including the material in this other CIP as agreed: https://github.com/cardano-foundation/CIPs/pull/25#issuecomment-760150596

**Comment by crptmppt** [comment]
@rphair the CIP is in 'Draft', there is much room to propose and add changes. As noted, Editors have limited bandwidth. Thanks for stepping up where needed, and I suggest you join the (new) CIP-Biweekly calls to be further involved - we're holding them publicly at https://www.crowdcast.io/cips-biweekly 
(Will strive to publish upcoming agendas a week in advance)

**Comment by rphair** [comment]
thank you @crptmppt for the confirmation & clarification about what it means to be a "draft" - that's very reassuring.  I appreciate the invitation, have subscribed to that channel & will be happy to join your next meeting 😎

**Comment by SebastienGllmt** [comment]
@rphair I'm kind of confused. There are two ways of merging our work:

1) Now that this PR is merged, your submit a follow-up PR that modifies this document that extends it with your proposal
2) Now that this PR is merged, your create a separate CIP that refers to this one and defines how you plan to extend it

My strong preference is on (1).

At any rate, both options required getting this PR to be merged first.

**Comment by rphair** [comment]
@SebastienGllmt as we left it, according the agenda resulting from our last conversation, my single-pool link spec was to be included in your own spec (step 3) before the payment spec was merged in as an approved CIP (step 4):

1. Simplify motivation
2. Remove multi-pool notation
3. Merge spec into Payment URI spec
4. Merge payment spec as approved draft CIP
5. Create new PR that adds the multi-pool notation and leave that as an open PR to the existing payment URI cip

I don't object to the change in plan, if that's what's it takes to keep the pool links included in this CIP going forward, but when observing step 4 happening before step 3 yesterday, after not hearing anything for 2 months, I think I was the one who was confused.

Also according to the original plan I would have only created one new PR: for adding the multi-pool spec into this CIP, since the single-pool spec would have been added before your own CIP pull request was merged.  Now I will be submitting two new PRs: one immediately for single-pool, and another down the road for multi-pool as we originally planned.

Thanks again for putting the key component through and allowing the URI scheme for this CIP to carry the stake pool links.  I would only insist that my upcoming pull request include a link to the other PR (https://github.com/cardano-foundation/CIPs/pull/25), which has now become a dead end, since it establishes a date on the original request as well as some helpful discussion.

**Comment by rphair** [comment]
The proposed changes have now been merged in via this new PR: https://github.com/cardano-foundation/CIPs/pull/61

**Pull request merged**